src/components/Primary-Button/Button.test.tsx

```diff
@@ -0,0 +1,17 @@
+import { render, screen } from '@testing-library/react';
+import { describe, it, expect } from 'vitest';
+
+import { PrimaryButton } from './Button';
+
+describe('PrimaryButton', () => {
```

Comment:
Current tests check rendering and disabled state. Add tests for: loading state (spinner present and content hidden/aria-busy set), fullWidth behavior, shape/variant rendering, and click handler invocation. Also prefer using getByRole for button queries (see previous comment).